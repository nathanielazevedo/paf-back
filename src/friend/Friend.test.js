/** @format */

import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import NavBar from "../nav/Navbar";
import { UserProvider } from "../testUtils";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <NavBar />
      </UserProvider>
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  let username = "Nate";
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <NavBar />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("displays expected text", function () {
  const { getAllByText } = render(
    <MemoryRouter>
      <UserProvider>
        <NavBar />
      </UserProvider>
    </MemoryRouter>
  );

  expect(getAllByText("Friends")[0]).toBeInTheDocument();
}); 