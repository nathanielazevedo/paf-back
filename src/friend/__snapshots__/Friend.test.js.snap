// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
  >
    <div
      class="navbar-container"
    >
      <a
        aria-current="page"
        class="navbar-logo active"
        href="/"
      >
        &lt;PAF/&gt;
      </a>
      <div
        class="menu-icon"
      >
        <i
          class="fas fa-bars"
        />
      </div>
      <ul
        class="nav-menu"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-links"
            href="/friends"
          >
            Friends
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-links"
            href="/sign-up"
          >
            Logout
          </a>
        </li>
      </ul>
    </div>
  </nav>
</DocumentFragment>
`;
